Block/Dom/RichText.php

```diff
@@ -35,6 +35,22 @@ public function __construct(
 
     public function fetchDocumentView(): string
     {
-        return PrismicRichText::asHtml($this->getContext(), $this->getLinkResolver(), [$this->htmlSerializer, 'serialize']);
+        $html = PrismicRichText::asHtml(
+            $this->getContext(),
+            $this->getLinkResolver(),
+            [$this->htmlSerializer, 'serialize']
+        );
+
+        $baseUrl = $this->_urlBuilder->getBaseUrl();
+        $findAndReplaceBaseUrl = [
+            'http://{{relative}}/',
+            'https://{{relative}}/',
+            'http://{{relative}}',
+            'https://{{relative}}',
+        ];
+
+        $html = str_replace($findAndReplaceBaseUrl, $baseUrl, $html);
```

Comment:
It seems you are duplicating the logic to replace the Base URL. Since you implement this once in an abstract block and the current file extends that abstract block, it seems to me you could deduplicate it by introducing a protected method in the abstract class and call that in both places.